src/models/data/data-set.ts

```diff
@@ -666,23 +757,29 @@ export const DataSet = types.model("DataSet", {
         });
       },
 
-      selectCases(caseIds: string[], select = true) {
-        if (select) self.attributeSelection.clear();
-        const ids: string[] = [];
-        caseIds.forEach(id => {
-          const pseudoCase = self.pseudoCaseMap[id];
-          if (pseudoCase) {
-            ids.push(...pseudoCase.childCaseIds);
-          } else {
-            ids.push(id);
-          }
-        });
-        ids.forEach(id => {
+      selectCases,
+
+      selectCells(cells: ICell[], select = true) {
+        if (select) {
+          self.attributeSelection.clear();
+          self.caseSelection.clear();
+        }
+
+        if (getAppConfig(self)?.getSetting("cellsSelectCases", "dataset")) {
```

Comment:
I'm a bit nervous about changing this, because I think it would require tiles to access `AppConfig` whenever they want to set the selected cells of a dataset. I feel like this will not be obvious to engineers and even for those who are aware it will be easy to forget. It makes sense to me that the `DataSet` would have direct access to its own settings. Is it really important to keep `AppConfig` out of `DataSet`?

Reply:
As a compromise, how about checking `AppConfig` in `afterCreate` and then setting a volatile property so that at least the `AppConfig` reference(s) are isolated to a single place in the code? Then at least it would be easier to eliminate if the need arises.

Reply:
I wasn't able to access `AppConfig` in `afterCreate`, but I was able to unify the `AppConfig` calls into a getter view.